app-creator/client/src/utils/helpers.ts

```diff
@@ -99,3 +119,47 @@ export function deepCloneTemplate(
 
   return clone;
 }
+
+/**
+ * Takes a snapshot of the current store and stores it in localStorage.
+ */
+export function storeSnapshotInLocalStorage(timestamp: number) {
+  try {
+    /**
+     * Saving current store snapshot as a string in localStorage so we can transfer data across.
+     * */
+    const storeSnapshot = JSON.stringify(deepCloneTemplate(store.getState()));
+
+    // Retrieve template snapshots object from local storage if it exists.
+    const templateSnapshots = localStorage.getItem(TEMPLATE_SNAPSHOTS);
```

Comment:
Is there a reason why you're not using `time` as a key directly in `localStorage`? If so, it's worth commenting about it.

Reply:
I feel its more organized to have the templates contained in one object. Keying directly on localStorage would save us some unnecessary parsing however.